### PR TITLE
RavelinJS v1.0.0

### PR DESCRIPTION
**Pull request by icio**
Once this PR is merged we will switch the default branch to be v1, keeping a v0 branch for the existing branch, and deleting the master branch. The repo requires reviewing as a whole, so I recommend checking [README.md](https://github.com/unravelin/ravelinjs/blob/v1.0.0-lib/README.md) and [CONTRIBUTING.md](https://github.com/unravelin/ravelinjs/blob/v1.0.0-lib/CONTRIBUTING.md).

##### Docs Checklist

- [ ] Wire up https://developer.ravelin.com/ up to this and the old versions.
- [ ] Include browser language in the server-side browser example
- [x] Reference v2/connect in README.
- [x] Write v0.1.0 -> v1.0.0 upgrade guide.
- [x] Update README with usage instructions, and development guide.
- [x] Add a copyright/license/attribution header to the ravelinjs built files. -- https://github.com/unravelin/ravelinjs/pull/34/commits/353c169487750861c50632fa625e751bb39bbd73

##### Checklist

- [x] Treat 19-digit pasted numbers as PANs. This is now safe because we require a data-rvn-pan attribute on inputs to be considered pans. 67b95496186abcfe130b82a626d4c3e83b8f11de
- [x] Test pasted value sanitisation with characters outside the ascii alphanumeric range. -- https://github.com/unravelin/ravelinjs/pull/34/commits/bcdb5d25bcd51e5fa734792ca7b3d759b2e7c4a8
- [x] Add error-sniffing to ravelin.core.id(). -- 7ef0cc46febd3affeb01a1ddc61ac1b761bc432c
- [x] Fix IE8 bugs with Encrypt.card error-report binding. -- https://github.com/unravelin/ravelinjs/pull/34/commits/5d4631e0e8399831bd47b8018d952701a845b9f4
- [x] Ditch Core.prototype.device until a later release. -- e8e9dbad7e03089e8fcd0bd60ae921870903f889
- [x] Create a core-only bundle. -- a86d5cfe57c67bb4025045736b9a7ed24f34eb57
- [x] Include version number in release filenames. -- 233964c7e8191bcc08b980c38c0b25b9576fa477
- [x] ~Replace lib/encryption-vendored with npm dependencies?~ sjcl isn't playing nicely with Rollup.
- [x] ~Discard keys that begin with sk_ or secret_key_.~ Not going to do this. It's better that we have it in our logs and know we need to rotate the keys than avoid reporting it's there and the secret key publication goes unnoticed.

### CODE REVIEW

File: lib/track.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
...DFE1\uDFE3]|\uD821[\uDC00-\uDFF7]|\uD823[\uDC00-\uDCD5\uDD00-\uDD08]|\uD82C[\uDC00-\uDD1E\uDD50-\uDD52\uDD64-\uDD67\uDD70-\uDEFB]|\uD82F[\uDC00-\uDC6A\uDC70-\uDC7C\uDC80-\uDC88\uDC90-\uDC99]|\uD835[\uDC00-\uDC54\uDC56-\uDC9C\uDC9E\uDC9F\uDCA2\uDCA5\uDCA6\uDCA9-\uDCAC\uDCAE-\uDCB9\uDCBB\uDCBD-\uDCC3\uDCC5-\uDD05\uDD07-\uDD0A\uDD0D-\uDD14\uDD16-\uDD1C\uDD1E-\uDD39\uDD3B-\uDD3E\uDD40-\uDD44\uDD46\uDD4A-\uDD50\uDD52-\uDEA5\uDEA8-\uDEC0\uDEC2-\uDEDA\uDEDC-\uDEFA\uDEFC-\uDF14\uDF16-\uDF34\uDF36-\uDF4E\uDF50-\uDF6E\uDF70-\uDF88\uDF8A-\uDFA8\uDFAA-\uDFC2\uDFC4-\uDFCB]|\uD838[\uDD00-\uDD2C\uDD37-\uDD3D\uDD4E\uDEC0-\uDEEB]|\uD83A[\uDC00-\uDCC4\uDD00-\uDD43\uDD4B]|\uD83B[\uDE00-\uDE03\uDE05-\uDE1F\uDE21\uDE22\uDE24\uDE27\uDE29-\uDE32\uDE34-\uDE37\uDE39\uDE3B\uDE42\uDE47\uDE49\uDE4B\uDE4D-\uDE4F\uDE51\uDE52\uDE54\uDE57\uDE59\uDE5B\uDE5D\uDE5F\uDE61\uDE62\uDE64\uDE67-\uDE6A\uDE6C-\uDE72\uDE74-\uDE77\uDE79-\uDE7C\uDE7E\uDE80-\uDE89\uDE8B-\uDE9B\uDEA1-\uDEA3\uDEA5-\uDEA9\uDEAB-\uDEBB]|\uD869[\uDC00-\uDEDD\uDF00-\uDFFF]|\uD86D[\uDC00-\uDF34\uDF40-\uDFFF]|\uD86E[\uDC00-\uDC1D\uDC20-\uDFFF]|\uD873[\uDC00-\uDEA1\uDEB0-\uDFFF]|\uD87A[\uDC00-\uDFE0]|\uD87E[\uDC00-\uDE1D]|\uD884[\uDC00-\uDF4A])/g, 'X');
```
Comment (sHesl) [inline lib/track.js:261]: v. nice!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+npm run build && npm run test:integration && npm run release
+```
+
+Locally, you'll be doing the following:
+
+* Edit code in ./lib.
+* Building code into the ./build directory with `npm run build`.
+* After building,
+    * Run unit tests locally with `npm run test:unit`.
+    * Authenticate with BrowserStack using `export BROWSERSTACK_USERNAME=x BROWSERSTACK_ACCESS_KEY=y`.
```
Comment (sHesl) [inline CONTRIBUTING.md]: Is it possible for us to stick a fixed key in 1password?

**Review by sHesl — COMMENTED** [review]
🏆 💯 🎖️ 

Can't say I've scrutinised every line of code, but you've been so detailed with this, didn't feel neccessary! I'll approve but if you can get eyes from some front-end eyes, that can't hurt as well.

Tested all the commands except the browserstack stuff, all worked first time.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
+# RavelinJS Contribution Guide
+
+If you're looking to change some code in RavelinJS, read this first.
+
+<!-- toc -->
+
+- [1. Familiarise yourself with the library.](#1-familiarise-yourself-with-the-library)
+- [2. Use the expected NodeJS v10.](#2-use-the-expected-nodejs-v10)
+- [3. Install a JSHint extention in your editor.](#3-install-a-jshint-extention-in-your-editor)
```
Comment (timphillips) [inline CONTRIBUTING.md]:
```suggestion
- [3. Install a JSHint extension in your editor.](#3-install-a-jshint-extension-in-your-editor)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+## 1. Familiarise yourself with the library.
+
+Have a read of the [README](README.md) to understand which parts of our library
+is meant to do what.
+
+## 2. Use the expected NodeJS v10.
+
+CI runs [circleci/node:10](.circleci/config.yml).
+
+## 3. Install a JSHint extention in your editor.
```
Comment (timphillips) [inline CONTRIBUTING.md]:
```suggestion
## 3. Install a JSHint extension in your editor.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+Promise](https://github.com/ysmood/yaku) in our polyfill bundle, so instead of
+handling rejections with `p.catch(function(err) { ... })`, use
+`p.then(undefined, function(err) { ... })`. Some IEs don't have
+Function.prototype.bind so use lib/util#bind.
+
+## 6. Prefer testing in unit tests.
+
+Unit tests in the test/*.test.js files have the benefit of running in a single
+page without needing server communication, so they're easy to run locally and
+very quick to run in CI. As a result, **attempt to write all new tests at unit tests**.
```
Comment (timphillips) [inline CONTRIBUTING.md]:
```suggestion
very quick to run in CI. As a result, **attempt to write all new tests as unit tests**.
```

**Review by timphillips — COMMENTED** [review]
Just a few typos I found in the docs.

**Pull request merged**